### PR TITLE
More readme fixes

### LINKED ISSUE #421: `ogx360` readme images are broken
https://github.com/Ryzee119/ogx360

image

### LINKED ISSUE #424: `dropbot-40-channel-HV-switching-board` readme images
|v1|v2|
|---|---|
|image|image|

### LINKED ISSUE #426: `GOSH-Board_UGM` readme images
|v1|v2|
|---|---|
|image|image|

### LINKED ISSUE #428: `uduino` readme images
https://github.com/intajay/uduino

|v1|v2|
|---|---|
|image|image|

### LINKED ISSUE #431: Emojis support in project description
This is nice to have.
|Kitspace|GitHub|
|---|---|
|image|image|

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
* Fixes #421
* Fixes #424
* Fixes #426
* Fixes #428 
* Closes #431

### CODE REVIEW

**Comment by kasbah** [comment]
Can we add some integration (or maybe even unit tests!?) for this functionality?

**Comment by AbdulrhmnGhanem** [comment]
I added e2e tests for them and will move them to the processor later as part of #448.

File: e2e/cypress/integration/readme.spec.js [hunk 1]
```diff
@@ -157,4 +157,63 @@ describe('Readme style', () => {
         cy.request($a[0].href).its('status').should('equal', 200)
       })
   })
+
+  it('should render :emoji: in readme and project description', () => {
```
Comment (kasbah) [inline e2e/cypress/integration/readme.spec.js:161]: Can this be an integration test instead?

File: e2e/cypress/integration/readme.spec.js [hunk 2]
```diff
@@ -157,4 +157,63 @@ describe('Readme style', () => {
...
+    cy.url({ timeout: 60_000 }).should('contain', `${username}/${repoName}`)
+    // Wait for the repo to finish processing, by checking the visibility of info-bar.
+    cy.get('[data-cy=info-bar]', { timeout: 60_000 }).should('be.visible')
+
+    // the project description isAdd modern xinput USB support to your Original 📺 🎮
+    cy.get('[data-cy=project-description]').should('contain.text', '📺 🎮')
+    cy.get('[data-cy=readme]').should('contain.text', '🤓')
+  })
+
+  it('should not modify the urls for GitHub Actions badges', () => {
```
Comment (kasbah) [inline e2e/cypress/integration/readme.spec.js:189]:
```suggestion
  it('preserves URLs for GitHub Actions badges', () => {
```

File: e2e/cypress/integration/readme.spec.js (hunk 1 above)
Comment (kasbah) [inline e2e/cypress/integration/readme.spec.js:161]:
```suggestion
  it('renders :emoji: in readme and project description', () => {
```

File: processor/src/utils.ts [hunk 3]
```diff
@@ -77,9 +77,14 @@ export function toGitHubRawUrl(url: string) {
   if (parsedUrl.hostname === 'github.com') {
     parsedUrl.hostname = 'raw.githubusercontent.com'
     const urlPath = parsedUrl.pathname.split('/')
-    // Remove `/blob/` or '/raw/' from the path.
-    parsedUrl.pathname = urlPath.slice(0, 3).concat(urlPath.slice(4)).join('/')
-    url = parsedUrl.toString()
+    // Avoid using  modifying github actions status badges.
```
Comment (kasbah) [inline processor/src/utils.ts:80]:
```suggestion
    // Avoid modifying github actions status badges.
```

**Review by kasbah — CHANGES_REQUESTED** [review]
Looks good, minor suggestions again. Please also correct the spelling mistake in your commit messages.

File: processor/src/tasks/writeKitspaceYaml.ts [hunk 4]
```diff
@@ -1,59 +1,59 @@
...
-  kitspaceYaml.summary = escapedSummary
-  return kitspaceYaml
+const Remarker = unified()
+  .use(remarkParse)
+  .use(remarkEmoji)
+  .use(remarkGfm)
+  .use(remarkRehype)
+  .use(rehypeHighlight)
+  .use(rehypeSanitize)
+  .use(rehypeStringify)
```
Comment (kasbah) [inline processor/src/tasks/writeKitspaceYaml.ts:54]: Do we need all of this processing. Full markdown support?

**Comment by AbdulrhmnGhanem** [comment]
> mutation of the input and also doing so off in an async way after the function returned.

Do you mean this line?
```ts
 Object.keys(kitspaceYaml.multi).forEach(async subProject => {
   linkifiedKitspaceYaml.multi[subProject] = await renderProjectSummary(kitspaceYaml.multi[subProject])
 })
```

File: processor/src/tasks/writeKitspaceYaml.ts (hunk 4 above)
Comment (AbdulrhmnGhanem) [reply processor/src/tasks/writeKitspaceYaml.ts:54]: I guess we can get rid of `remarkGfm`.

**Comment by kasbah** [comment]
Yeah, that creates an async function for each key in the yaml but doesn't await or return it. Just before that you had

```ts
const linkifiedKitspaceYaml = kitspaceYaml
```

Which references the same object and then you mutate it with the async functions.

The scenario where this could have caused problems is if we had another sub-task that shares the `kitspaceYaml` with `writeKitspaceYaml` and wanted to use the summary for something where it really shouldn't be html. 


Overall I'm wondering if this is really the right place to render the description. Shouldn't the frontend just handle it? It's fine for now... let's not spend time on it but something to think about for the future.

File: processor/src/tasks/writeKitspaceYaml.ts (hunk 4 above)
Comment (kasbah) [reply processor/src/tasks/writeKitspaceYaml.ts:54]:
```suggestion
  .use(remarkParse)
  .use(remarkEmoji)
  .use(remarkRehype)
  .use(rehypeSanitize)
  .use(rehypeStringify)
```

**Comment by kasbah** [comment]
`s/format in project reamde & summary/format in project readme & summary/`

File: processor/src/tasks/writeKitspaceYaml.ts [hunk 5]
```diff
@@ -1,59 +1,57 @@
...
-  const matches = linkify.match(escapedSummary)
-
-  if (matches) {
-    for (const match of matches) {
-      // Use https by default
-      const url = new URL(match.url)
-      if (!match.schema) {
-        url.protocol = 'https:'
+    const rendered = { multi: {} }
+    for (const key of kitspaceYaml.multi) {
```
Comment (kasbah) [inline processor/src/tasks/writeKitspaceYaml.ts:35]:
```suggestion
    for (const key of Object.keys(kitspaceYaml.multi)) {
```
e6f236539a1d1ddf02cb6d1884407454-3427410689

File: processor/src/tasks/writeKitspaceYaml.ts [hunk 6]
```diff
@@ -1,59 +1,57 @@
...
+      const subProject = kitspaceYaml.multi[key]
+      rendered.multi[subProject] = {
```
Comment (kasbah) [inline processor/src/tasks/writeKitspaceYaml.ts:37]:
```suggestion
      rendered.multi[key] = {
```

File: processor/src/tasks/writeKitspaceYaml.ts [hunk 7]
```diff
@@ -1,59 +1,57 @@
-import escape from 'escape-html'
-import LinkifyIt from 'linkify-it'
 import path from 'node:path'
 
+import { unified } from 'unified'
+import rehypeHighlight from 'rehype-highlight'
+import rehypeSanitize from 'rehype-sanitize'
+import rehypeStringify from 'rehype-stringify'
+import remarkEmoji from 'remark-emoji'
+import remarkGfm from 'remark-gfm'
```
Comment (kasbah) [inline processor/src/tasks/writeKitspaceYaml.ts:8]: It seems remarkGfm is actually handling the auto-linking so should be re-added.

**Pull request merged**